File: morphac/math/numeric/python/test/integrator_type_test.py
import pytest

from morphac.math.numeric import IntegratorType


@pytest.fixture()
def generator_integrator_type():

    euler_type = IntegratorType.EULER_INTEGRATOR
    mid_point_type = IntegratorType.MID_POINT_INTEGRATOR
    rk4_type = IntegratorType.RK4_INTEGRATOR

    return euler_type, mid_point_type, rk4_type


def test_type_equality(generator_integrator_type):
    euler_type, mid_point_type, rk4_type = generator_integrator_type

    assert euler_type == IntegratorType.EULER_INTEGRATOR
    assert mid_point_type == IntegratorType.MID_POINT_INTEGRATOR
    assert rk4_type == IntegratorType.RK4_INTEGRATOR

    assert euler_type is IntegratorType.EULER_INTEGRATOR
    assert mid_point_type is IntegratorType.MID_POINT_INTEGRATOR
    assert rk4_type is IntegratorType.RK4_INTEGRATOR

    assert euler_type != IntegratorType.MID_POINT_INTEGRATOR
    assert mid_point_type != IntegratorType.RK4_INTEGRATOR
    assert rk4_type != IntegratorType.EULER_INTEGRATOR

    assert euler_type is not IntegratorType.MID_POINT_INTEGRATOR
    assert mid_point_type is not IntegratorType.RK4_INTEGRATOR
    assert rk4_type is not IntegratorType.EULER_INTEGRATOR

File: morphac/math/numeric/test/integrator_type_test.cc
#include "gtest/gtest.h"

#include "math/numeric/include/integrator.h"

namespace {

using morphac::math::numeric::IntegratorType;

TEST(IntegratorTypeTest, Equality) {
  IntegratorType euler_type{IntegratorType::kEulerIntegrator};
  IntegratorType mid_point_type{IntegratorType::kMidPointIntegrator};
  IntegratorType rk4_type{IntegratorType::kRK4Integrator};

  ASSERT_TRUE(euler_type == IntegratorType::kEulerIntegrator);
  ASSERT_TRUE(mid_point_type == IntegratorType::kMidPointIntegrator);
  ASSERT_TRUE(rk4_type == IntegratorType::kRK4Integrator);

  ASSERT_TRUE(euler_type != IntegratorType::kMidPointIntegrator);
  ASSERT_TRUE(mid_point_type != IntegratorType::kRK4Integrator);
  ASSERT_TRUE(rk4_type != IntegratorType::kEulerIntegrator);
}

}  // namespace

int main(int argc, char** argv) {
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}

